zerver/views/auth.py

```diff
@@ -904,6 +905,15 @@ def password_reset(request: HttpRequest, **kwargs: Any) -> HttpResponse:
                                  password_reset_form=ZulipPasswordResetForm,
                                  post_reset_redirect='/accounts/password/reset/done/')
 
+@has_request_variables
+def auth_session_with_api_key(request: HttpRequest, user_profile: UserProfile,
+                              next: str=REQ(default='')) -> HttpResponse:
+    django_login(request, user_profile, 'zproject.backends.ZulipDummyBackend')
```

Comment:
We'll want to validate the `next` parameter for open redirect issues like we do for other endpoints that support it.